File: user-service/src/main/java/socialapp/userservice/service/RelationService.java
package socialapp.userservice.service;

import socialapp.userservice.model.dto.UserRelationsDto;

public interface RelationService {

    void deleteBlock(Long id);

    void deleteSubscription(Long id);

    void deleteFriendship(Long id, Long by);

    void blockUser(Long sender,Long receiver);

    void subscribeUser(Long sender,Long receiver);

    void addFriendship(Long sender,Long receiver);

    UserRelationsDto findUserRelations(Long userId);
}

File: kt-user-service/src/main/kotlin/socialapp/ktuserservice/service/RelationService.kt
package socialapp.ktuserservice.service

import socialapp.ktuserservice.model.dto.UserRelationsDto

interface RelationService {

    fun deleteBlock(id: Long)

    fun deleteSubscription(id: Long)

    fun deleteFriendship(id: Long, by: Long)

    fun blockUser(sender: Long, receiver: Long)

    fun subscribeUser(sender: Long, receiver: Long)

    fun addFriendship(sender: Long, receiver: Long)

    fun findUserRelations(userId: Long): UserRelationsDto
}